lib/Transforms/LoopFusion.cpp

```diff
@@ -1491,17 +1515,20 @@ struct GreedyFusion {
           // Skip if 'srcNode' is not a loop nest.
           if (!isa<AffineForOp>(srcNode->op))
             continue;
-          // Skip if 'srcNode' has more than one store to any memref.
-          // TODO(andydavis) Support fusing multi-output src loop nests.
-          if (srcNode->stores.size() != 1)
+          // Skip if 'srcNode' has more than one store to a function-local
+          // live-out.
+          // TODO(andydavis) Support more generic multi-output src loop nests
+          // fusion.
+          auto srcStoreOp = mdg->getUniqueStoreToLiveOut(srcNode);
```

Comment:
Shouldn't this store be on 'memref'? Since you are actually checking getOutEdgeCount != 1, you've made sure to eliminate the "no store case" on memref. So, if this finds a store, it will have to be on 'memref'.